pkg/mpc-examples/contracts/04-pause-unpause-controller/PauseUnpauseControllerFactory.sol

```diff
@@ -0,0 +1,144 @@
+
+    // This struct is a subset of IManagedPoolFactory.NewPoolParams which omits arguments
+    // that this factory will override and are therefore unnecessary to provide. It will
+    // ultimately be used to populate IManagedPoolFactory.NewPoolParams.
+    struct MinimalPoolParams {
+        string name;
+        string symbol;
+        IERC20[] tokens;
+        uint256[] normalizedWeights;
+        uint256 swapFeePercentage;
+        bool swapEnabledOnStart;
+        uint256 managementAumFeePercentage;
+        uint256 aumFeeId;
+    }
+
+    event ControllerCreated(address indexed controller, bytes32 poolId);
+    event Disabled();
+
+    constructor(IVault vault, address factory) {
+        managedPoolFactory = IManagedPoolFactory(factory);
+        balancerVault = vault;
+    }
+
+    /// === Getters === ///
+
+    function isDisabled() public view returns (bool) {
+        return _disabled;
+    }
+
+    function getLastCreatedPool() external view returns (address) {
+        return _lastCreatedPool;
+    }
+
+    /// === Setters === ///
+
+    function create(MinimalPoolParams calldata minimalParams, address controllerOwner) external {
+        // checks
+        _ensureEnabled();
+
+        bytes32 controllerSalt = bytes32(_nextControllerSalt);
+        _nextControllerSalt += 1;
+
+        bytes memory controllerCreationCode = abi.encodePacked(
+            type(PauseUnpauseController).creationCode,
+            abi.encode(balancerVault, controllerOwner) //constructor args
+        );
+
+        address expectedControllerAddress = Create2.computeAddress(controllerSalt, keccak256(controllerCreationCode));
+
+        // The asset managers have the ability to manage poolTokens. Since this controller
+        // is focused on a secure pause/unpausing, allowing to freely pass asset managers
+        // during pool creation could undermine the narrative that a paused pool does not have
+        // any "token movements". Passing the controller as asset manager (which does not have
+        // any logic for that purpose) confirms that narrative for this example.
+        address[] memory assetManagers = new address[](minimalParams.tokens.length);
+        for (uint256 i = 0; i < assetManagers.length; i++) {
+            assetManagers[i] = expectedControllerAddress;
+        }
+
+        IManagedPoolFactory.NewPoolParams memory fullParams;
+        fullParams.name = minimalParams.name;
+        fullParams.symbol = minimalParams.symbol;
+        fullParams.tokens = minimalParams.tokens;
+        fullParams.normalizedWeights = minimalParams.normalizedWeights;
+        fullParams.assetManagers = assetManagers;
+        fullParams.swapFeePercentage = minimalParams.swapFeePercentage;
+        fullParams.swapEnabledOnStart = minimalParams.swapEnabledOnStart;
+        fullParams.mustAllowlistLPs = false;
+        fullParams.managementAumFeePercentage = minimalParams.managementAumFeePercentage;
+        fullParams.aumFeeId = minimalParams.aumFeeId;
```

Comment:
Add comments for arguments that are set by this controller, rather than populated from `minimalParams.*`